src/arithmetic.jl

```diff
@@ -0,0 +1,501 @@
+# This file is part of the TaylorSeries.jl Julia package, MIT license
+#
+# Luis Benet & David P. Sanders
+# UNAM
+#
+# MIT Expat license
+#
+
+# Arithmetic operations: +, -, *, /
+
+## Equality ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
+        function =={T<:Number}(a::$T{T}, b::$T{T})
+            a, b = fixorder(a, b)
+            @simd for i in eachindex(a.coeffs)
+                @inbounds (a.coeffs[i] == b.coeffs[i]) || return false
+            end
+            true
+        end
+    end
+end
+
+==(a::HomogeneousPolynomial, b::HomogeneousPolynomial) = a.coeffs == b.coeffs
+
+
+function iszero{T}(a::HomogeneousPolynomial{T})
+    @simd for i in eachindex(a.coeffs)
+        @inbounds (a.coeffs[i] == zero(T)) || return false
```

Comment:
Sorry, I did something wrong regarding my check with your proposal; indeed tests pass. Regarding performance (`perf/fateman.jl`) it is slightly degraded (few percent), the number of allocations grows as well as the total memory used grows. So the current implementation, though not so nice, is a bit better.